app/service/agent_svc.py

```diff
@@ -73,13 +73,14 @@ def __init__(self, untrusted_timer):
             if agent[0]['trusted']:
                 update_data['last_trusted_seen'] = now
             await self.get_service('data_svc').update('core_agent', 'paw', paw, data=update_data)
-            return agent[0]
         else:
             queued = dict(last_seen=now, paw=paw, platform=platform, server=server, host_group=group,
                           location=location, architecture=architecture, pid=pid, ppid=ppid,
-                          trusted=True, last_trusted_seen=now, sleep=sleep)
+                          trusted=True, last_trusted_seen=now, sleep_min=sleep, sleep_max=sleep)
             await self.get_service('data_svc').create_agent(agent=queued, executors=executors)
-            return (await self.get_service('data_svc').explode_agents(criteria=dict(paw=paw)))[0]
+            agent = await self.get_service('data_svc').explode_agents(criteria=dict(paw=paw))
+        agent[0]['sleep'] = self.jitter_range(agent[0]['sleep_min'], agent[0]['sleep_max'])
+        return agent[0]
```

Comment:
this looks a little messy

Reply:
I can hide the sleep in the explode_agents function

Reply:
Made much more sense there.